src/devices/Ssd1306/Ssd1306.cs

```diff
@@ -0,0 +1,57 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Iot.Device.Ssd1306.Command;
+using System;
+using System.Device.I2c;
+
+namespace Iot.Device.Ssd1306
+{
+    public class Ssd1306 : IDisposable
+    {
+        private I2cDevice _i2cDevice;
+
+        public Ssd1306(I2cDevice i2cDevice)
+        {
+            _i2cDevice = i2cDevice;
+        }
+
+        public void SendCommand(ICommand command, bool continuation = false)
+        {
+            byte controlByte = 0x00;
+            byte[] commandBytes = command.GetBytes();
```

Comment:
I'm not an expert with Span but I believe we might be able to get away with not needing the two arrays and the copies if we just use Span instead. @ahsonkhan do you mind taking a look at this? Is it possible to do something like:

```c#
Span<byte> span = stackalloc byte[command.Length + 1];
span[0] = 0x00;
span[1] = command.GetBytes();
```

Reply:
It is **not** recommended that you`stackalloc` an unconstrained amount based on user passed data. The user can pass in a command where `.GetBytes` returns MB/GB of bytes which would blow up the stack. The recommended pattern is based around a constant stack size threshold.

```C#
const int StackThreshold = 256;

byte[] commandBytes = command.GetBytes();

Span<byte> buffer = commandBytes.Length < StackThreshold ?
   stackalloc byte[commandBytes.Length + 1] :
   new byte [commandBytes.Length + 1];

commandBytes.CopyTo(buffer.Slice(1));

if (continuation)
{
   buffer[0] = 0x80;
}

_i2cDevice.Write(buffer);
```

Reply:
As an aside, don't we need argument validation like `.GetBytes` doesn't return `null`? What happens if I return an empty array?

Reply:
Question on input size, how big of an array can we reasonable expect `GetBytes()` to return here? Not sure if it would be useful to leverage the array pool.

Reply:
All the Commands that are provided by the binding won't return more than 20 bytes (@shaggygi correct me if I'm wrong) which is why I thought that stackalloc wouldn't be too bad. Of course a user could create their own ICommand that returns way more bytes and can run into those  kind of issues but I'm not sure we should try to guard against that case. Most common usage here would be of the likes of:
```c#
ssd1306.SendCommand(new SetDisplayOff());
ssd1306.SendCommand(new SetDisplayClockDivideRatioOscillatorFrequency(0x00, 0x08));
ssd1306.SendCommand(new SetMultiplexRatio(0x1F));
```
Where those commands return around 2 or 3 bytes at most on the GetBytes call.

Reply:
Thanks for the input @ahsonkhan.  I'm not sure if GetBytes was the best naming in this case as some might think it's trying to get data from the device/bus.  It is only used to get the respective data that makes up a command.  @joperezr is correct, as a command is usually a handful of bytes (1 - 5) that are sent to the display.  I don't believe a user would create their own command ,and if so, it would most likely require the one byte for Value (although they still could return an empty array as you mentioned).

Hope this helps.

Reply:
> I don't believe a user would create their own command ,and if so, it would most likely require the one byte for Value (although they still could return an empty array as you mentioned).

In that case, it is likely even more useful to use constrained stackalloc rather than allocating small 1-5 byte arrays, imo. Up to you.

You should still do `null` / empty array checks though.